modules/programs/awscli.nix

```diff
@@ -3,7 +3,14 @@
 let
   cfg = config.programs.awscli;
   iniFormat = pkgs.formats.ini { };
-
+  settingsPath =
+    if config.programs.awscli.settings.path != ""
```

Comment:
Hmm, I had a quick look and there doesn't seem like there is a `path` field in `~/.aws/config` configuration file? Similarly, in the `~/.aws/credentials` file. The entries inside `programs.awscli.settings` and `programs.awscli.credentials` should match actual options honored by the aws-cli tool.

Reply:
I.e., if you want to control the path of these files, then it should be set outside the `settings` and `credentials` options.

Reply:
Yeah, I also realized after submitting the PR that it'd make it backwards incompatible, so instead of altering the `settings` and `credentials` options it'd be better to create an extra option to specify the path of these files.

I also thought about creating an option to `xdg` called `honor` or something, that would apply the required variables to make the tools all comply with XDG Base Directory Specification. It's outside the scope of this PR though.

Reply:
Yeah, having separate path options work much better.

Btw, there is already a [`home.preferXdgDirectories`](https://nix-community.github.io/home-manager/options.xhtml#opt-home.preferXdgDirectories) option for the purpose of forcing programs to use XDG directories.